packages/test/test-end-to-end-tests/src/test/loadModes.spec.ts

```diff
@@ -0,0 +1,375 @@
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		assert.notEqual(dataObject1.value, dataObject2.value, "counter values should not be equal");
+		assert.notEqual(
+			container1.deltaManager.lastSequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container sequence numbers should not be equal",
+		);
+		assert.strictEqual(
+			initialValue,
+			dataObject2.value,
+			"sharedCounter2 should still be the initial value",
+		);
+	});
+
+	it("Can load a paused container at a specific sequence number", async () => {
+		// Min 5 ops, max 15 ops
+		const numIncrement = Math.round(Math.random() * 10) + 5;
+		for (let i = 0; i < numIncrement; i++) {
+			dataObject1.increment();
+		}
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		// Record sequence number we want to pause at, and the expected value at that sequence number
+		const sequenceNumber = container1.deltaManager.lastSequenceNumber;
+		const expectedValue = dataObject1.value;
+
+		const headers: IRequestHeader = {
+			[LoaderHeader.loadMode]: {
+				pauseAfterLoad: true,
+				opsBeforeReturn: "sequenceNumber",
+			},
+			[LoaderHeader.sequenceNumber]: sequenceNumber,
+		};
+		const container2 = await loadContainer(
+			container1.resolvedUrl,
+			testDataObjectFactory,
+			headers,
+		);
+		const dataObject2 = await requestFluidObject<TestDataObject>(container2, "default");
+
+		assert.strictEqual(
+			sequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container2 should be at the specified sequence number",
+		);
+		assert.strictEqual(
+			expectedValue,
+			dataObject2.value,
+			"sharedCounter2 should still be the expected value",
+		);
+		assert.strictEqual(dataObject1.value, dataObject2.value, "counter values should be equal");
+
+		for (let i = 0; i < numIncrement; i++) {
+			dataObject1.increment();
+		}
+		await loaderContainerTracker.ensureSynchronized(container1);
+
+		assert.notEqual(dataObject1.value, dataObject2.value, "counter values should not be equal");
+		assert.notEqual(
+			container1.deltaManager.lastSequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container sequence numbers should not be equal",
+		);
+
+		assert.strictEqual(
+			sequenceNumber,
+			container2.deltaManager.lastSequenceNumber,
+			"container2 should still be at the specified sequence number",
+		);
+	});
+
+	describe("Expected error cases", () => {
+		it("Throw if sequence number not provided", async () => {
+			const headers: IRequestHeader = {
+				[LoaderHeader.loadMode]: {
+					opsBeforeReturn: "sequenceNumber",
+				},
+			};
+			try {
+				await loadContainer(container1.resolvedUrl, testDataObjectFactory, headers);
+				assert.fail("Did not throw expected error");
+			} catch (e: any) {
+				const expectedError = "sequenceNumber must be set to a non-negative integer";
+				assert.ok(e.message);
+				assert.strictEqual(e.message, expectedError, "Did not get expected error message");
+			}
```

Comment:
Consider using assert.rejects() to express this more explicitly and save some legwork here and instances below
https://nodejs.org/api/assert.html#assertrejectsasyncfn-error-message

I always have to fiddle with the syntax to remember exactly how it works, but something like this:

```suggestion
			assert.rejects(
				loadContainer(container1.resolvedUrl, testDataObjectFactory, headers),
				{ message: "sequenceNumber must be set to a non-negative integer" },
			);
```